server/src/main/java/com/objectcomputing/checkins/services/checkins/CheckInServicesImpl.java

```diff
@@ -30,50 +35,67 @@ public class CheckInServicesImpl implements CheckInServices {
     private final MemberProfileRepository memberRepo;
     private final CurrentUserServices currentUserServices;
     private final RoleServices roleServices;
+    private final PermissionServices permissionServices;
 
     public CheckInServicesImpl(CheckInRepository checkinRepo,
                                MemberProfileRepository memberRepo,
-                               CurrentUserServices currentUserServices, RoleServices roleServices) {
+                               CurrentUserServices currentUserServices,
+                               RoleServices roleServices, PermissionServices permissionServices) {
         this.checkinRepo = checkinRepo;
         this.memberRepo = memberRepo;
         this.currentUserServices = currentUserServices;
         this.roleServices = roleServices;
+        this.permissionServices = permissionServices;
+    }
+
+    public Boolean hasElevatedAccessPermission(@NotNull UUID memberId) {
+        boolean hasElevatedAccess = false;
+        List<Permission> userPermissions = permissionServices.findUserPermissions(memberId);
+        System.out.println("USER PERMISSIONS " + userPermissions);
+        if (userPermissions.size() > 0) {
+            hasElevatedAccess = userPermissions.stream().map(Permission::getPermission).anyMatch(str -> str.equals(Permissions.ELEVATED_ACCESS.name()));
+            LOG.debug("Member has elevated access permisson: {}", hasElevatedAccess);
+        }
+        return hasElevatedAccess;
     }
 
     @Override
     public Boolean accessGranted(@NotNull UUID checkinId, @NotNull UUID memberId) {
         Boolean grantAccess = false;
 
-        MemberProfile memberTryingToGainAccess = memberRepo.findById(memberId).orElseThrow(() -> {
+        memberRepo.findById(memberId).orElseThrow(() -> {
             throw new NotFoundException(String.format("Member %s not found", memberId));
         });
         CheckIn checkinRecord = checkinRepo.findById(checkinId).orElseThrow(() -> {
             throw new NotFoundException(String.format("Checkin %s not found", checkinId));
         });
```

Comment:
Instead of being supplied with a function that throws an exception, `orElseThrow` should be given an exception:
```
call.orElseThrow(() -> new NotFoundException("message"));
```

This occurs a few times in this file, not due to your changes, but it seems worth changing. (see also: line 86)